apps/catalog/catalog-e2e/src/integration/dashboard/deal-seller-negotiatiation.spec.ts

```diff
@@ -0,0 +1,283 @@
+    get('territories').click();
+    get('Latin America').click();
+    escapeKey();
+    //taking off free TV
+    get('medias').should('contain', 'TV');
+    get('medias').click();
+    get('freeTv').click();
+    escapeKey();
+    //adding one month before and after
+    get('dateFrom').clear().type(seller.term.duration.from.toLocaleDateString('en-US'));
+    get('dateTo').clear().type(seller.term.duration.to.toLocaleDateString('en-US'));
+    //changing to exclusive
+    get('exclusivity').click();
+    get('exclusive').click();
+    //add french version
+    get('add-version').click();
+    get('languages').find('input').type('f');
+    get('option_french').click();
+    get('save-language').click();
+    get('subtitle').click();
+    get('dubbed').click();
+    get('caption').click();
+    //save & submit
+    assertTableRowData(0, [
+      seller.term.duration.from.toLocaleDateString('en-US'),
+      seller.term.duration.to.toLocaleDateString('en-US'),
+      'Latin America, Europe',
+      'Pay TV, Pay Per View',
+      'Yes',
+      'French: Dubs, Subs, CC',
+    ]);
+    get('row-save').click();
+    acceptCookies(); //it hides the submit button
+    get('submit').click();
+    get('modal').should('exist');
+    get('confirm').click();
+    snackbarShould('contain', 'Your counter offer has been sent');
+    get('status-tag').should('contain', 'In Negotiation');
+    //sales main page
+    checkMainSalePage('negotiation');
+    //emails & notifications creation
+    checkConfirmationEmails('negotiation');
+    checkNotification('seller', 'negotiation');
+    //buyer's offers main page
+    connectOtherUser(buyer.user.email);
+    cy.visit('/c/o/marketplace/offer');
+    get('row_0_col_6').should('contain', 'In Negotiation');
+    //offer page
+    get('row_0_col_0').click();
+    get('status-tag').should('contain', 'In Negotiation');
+    get('row_0_col_4').should('contain', 'In Negotiation');
+    checkNotification('buyer', 'negotiation');
+    //contract page
+    cy.visit(`c/o/marketplace/offer/${offer.id}/${saleContract.id}`);
+    get('offer-price').should('contain', '€15,000.00');
+    assertTableRowData(0, [
+      seller.term.duration.from.toLocaleDateString('en-US'),
+      seller.term.duration.to.toLocaleDateString('en-US'),
+      'Latin America, Europe',
+      'Pay TV, Pay Per View',
+      'Yes',
+      'French (Dubs, Subs, CC)',
+    ]);
+    //accept
+    get('accept').click();
+    get('terms').click();
+    get('confirm').click();
+    snackbarShould('contain', `You accepted contract for ${seller.movie.title.international}`);
+    get('status-tag').should('contain', 'Accepted');
+    //checking on seller's side
+    connectOtherUser(seller.user.email);
+    cy.visit('c/o/dashboard/sales');
+    get('row_0_col_5').should('contain', '€15,000.00');
+    get('row_0_col_6').should('contain', 'Accepted');
+  });
+});
+
+//* functions
+
+function checkConfirmationEmails(decision: 'accepted' | 'declined' | 'negotiation') {
```

Comment:
looks like you can use `ContractStatus`
```suggestion
function checkConfirmationEmails(contractStatus: ContractStatus) {
```